res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,426 @@
+        if (isSyncLeader) {
+        	var otherDeckNr = deckNr === 1 ? 2 : 1;
+        	engine.setValue("[Channel" + otherDeckNr + "]", keyString, false);
+        }
+        engine.setValue(controlString, keyString, isSyncLeader);
+        Mixage.toggleLED(isSyncLeader ? 1 : 0, controlString, keyString);
+    }*/
+};
+
+// The FX ON button that toggles routing channel 1/2 through effects
+Mixage.handleEffectChannelOn = function(channel, control, value, _status, _group) {
+    // calculate effect unit number from MIDI control. 0x08 controls unit 1, 0x16 unit 2
+    var unitNr = control === 0x08 ? 1 : 2;
+    // react only on first message / keydown
+    if (value === 0x7F) {
+        // check and toggle enablement
+        Mixage.effectRackEnabled[unitNr] = !Mixage.effectRackEnabled[unitNr];
+        var isEnabled = Mixage.effectRackEnabled[unitNr];
+        // update controls
+        var keyString = "group_[Channel" + unitNr + "]_enable";
+        engine.setValue("[EffectRack1_EffectUnit1]", keyString, isEnabled && Mixage.effectRackSelected[unitNr][0]);
+        engine.setValue("[EffectRack1_EffectUnit2]", keyString, isEnabled && Mixage.effectRackSelected[unitNr][1]);
+        Mixage.toggleLED(isEnabled ? 1 : 0, "[Channel" + unitNr + "]", "fx_on");
+    }
+};
+
+// The FX SEL button that selects which effects are enabled for channel 1/2
+Mixage.handleEffectChannelSelect = function(channel, control, value, _status, _group) {
+    // calculate effect unit number from MIDI control. 0x07 controls unit 1, 0x15 unit 2
+    var unitNr = control === 0x07 ? 1 : 2;
+    // react only on first message / keydown
+    if (value === 0x7F) {
+        // check and toggle select state
+        var selected1 = Mixage.effectRackSelected[unitNr][0];
+        var selected2 = Mixage.effectRackSelected[unitNr][1];
+        if (selected1 && selected2) {
+            selected1 = true;
+            selected2 = false;
+        } else if (selected1) {
+            selected1 = false;
+            selected2 = true;
+        } else {
+            selected1 = true;
+            selected2 = true;
+        }
+        Mixage.effectRackSelected[unitNr][0] = selected1;
+        Mixage.effectRackSelected[unitNr][1] = selected2;
+        // update controls
+        var isEnabled = Mixage.effectRackEnabled[unitNr];
+        var keyString = "group_[Channel" + unitNr + "]_enable";
+        engine.setValue("[EffectRack1_EffectUnit1]", keyString, isEnabled && Mixage.effectRackSelected[unitNr][0]);
+        engine.setValue("[EffectRack1_EffectUnit2]", keyString, isEnabled && Mixage.effectRackSelected[unitNr][1]);
+    }
+};
+
+Mixage.handleEffectDryWet = function(channel, control, value, _status, _group) {
+    // calculate effect unit number from MIDI control. 0x21 controls unit 1, 0x25 unit 2
+    var unitNr = control === 0x21 ? 1 : 2;
+    // Control centers on 0x40 (64), calculate difference to that value and multiply by 4
+    var diff = (value - 64) / 10.0;
```

Comment:
This doesn't seem accurate.

Reply:
Tradeoff. More that half a knob turn is slow and not necessary for dry/wet and super imo. I've increased the steps to 16. Imo this should be accurate enough and feels pretty good.